### ISSUE TITLE
Consistently lower argsort to CUB for all primitive data types

### ISSUE THREAD

**Issue by jstuecker** [jax-ml/jax#34912]
Currently using argsort feels like a performance lottery, since it appears to work well for some data types, but performs extremely poorly for others. Consider e.g. the following example

```python
import time
import jax
import jax.numpy as jnp

@jax.jit
def argsort64(indices):
    with jax.enable_x64():
        return jnp.argsort(indices.astype(jnp.int64))

@jax.jit
def argsort32(indices):
    return jnp.argsort(indices)

indices = jax.random.randint(jax.random.key(0), 1024**3, 0, 1024)

t0 = time.perf_counter()
argsort32.lower(indices).compile()
t1 = time.perf_counter()
argsort64.lower(indices).compile()
t2 = time.perf_counter()
for i in range(10):
    jax.block_until_ready(argsort32(indices))
t3 = time.perf_counter()
for i in range(10):
    jax.block_until_ready(argsort64(indices))
t4 = time.perf_counter()

print(f"argsort32: compile:{t1-t0:.3f} run:{(t3-t2)/10.:.3f}")
print(f"argsort64: compile:{t2-t1:.3f} run:{(t4-t3)/10.:.3f}")
```
gives:
```
argsort32: compile:0.038 run:0.062
argsort64: compile:3.405 run:5.304
```
(tested with NVIDIA A100-SXM-64GB, jax 0.8.2 and CUDA 12.9)
So switching from 32 to 64 bit integers results in a performance loss of a factor 100 in both compile and run time... I think this is likely, because int32 lower to a CUB radix sort and int64 don't... (For reference, the expected performance drop should be roughly a factor 4: https://wmaxey.github.io/cccl/cub/api/structcub_1_1DeviceRadixSort.html)
It would be great to have consistently good performance across the board by lowering the sorting of any primitive data types to a CUB radix sort. Beyond that it would be very nice to have the option to define the data type explicitly so that one is not forced to use a 100x slower sort, just because one enabled x64 to use double precision numbers in another part of the code...
Related issues:
https://github.com/jax-ml/jax/issues/27004
https://github.com/openxla/xla/issues/35587
https://github.com/jax-ml/jax/issues/10434#issuecomment-3145594426

**Comment by jakevdp** [comment]
Thanks for the report! This is an XLA-level thing, so it seems that pinging https://github.com/openxla/xla/issues/35587 would be the best way to get the fix at the XLA level.

> Beyond that it would be very nice to have the option to define the data type explicitly

This seems like a reasonable request whether or not the performance issue persists: we could add a `dtype` argument to `jnp.argsort` that would let the user specify this.
